botocore/session.py

```diff
@@ -945,6 +958,7 @@ def create_client(
                 access_key=aws_access_key_id,
                 secret_key=aws_secret_access_key,
                 token=aws_session_token,
```

Comment:
Right now, this looks like we are just silently dropping `token` and `account_id` if they are set without `aws_access_key_id` and `aws_secret_access_key`, is that correct?

That's probably fine- we haven't had any customer complaints yet that I know of, but it makes sense to me to throw a debug statement here that we are discarding the `token` or `account_id`(unless they can get consumed some other way?)